### PR TITLE
[doc] Clarify HMAC and KMAC endianness settings

### LINKED ISSUE #6251: [doc] Clarify the behavior of the `hmac.CFG.endian_swap` and `kmac.CFG.msg_endianness` flags
The `hmac.CFG.endian_swap` and `kmac.CFG.msg_endianness` flags both disable or enable byte swapping on values written into the appropriate `MSG_FIFO`. However from a software perspective the technical specifications do not make clear exactly what the behavior is when values smaller than a word are written into the FIFO. It would be good to clarify this. Similarly it would be nice to unambiguously specify the behavior of unaligned writes and that the order of the writes themselves is affected by the flag.

### `hmac.CFG.endian_swap`

**Current text:** Convert TL-UL wdata[31:0] to big-endian style {w[7:0], w[15:8], .. }

**Suggested text:** Message byte swap disable. If 0 each individual multi-byte value, regardless of its alignment, written to `MSG_FIFO` will be added to the message in big-endian byte order. If 1 the value will be added to the message in little-endian byte order. Writes will not be reordered with regard to one another. A message written to `MSG_FIFO` one byte at a time will therefore not be affected by this setting.

### `kmac.CFG.msg_endianness`

**Current text:** Message Endianness. If 1, Convert TL-UL write data[31:0] to {d[7:0], d[15:8], ...}0: Little-endian. Keep input value as it is 1: Big-endian. Convert incoming value

**Suggested text:** Message byte swap enable. If 1 each individual multi-byte value, regardless of its alignment, written to `MSG_FIFO` will be added to the message in big-endian byte order. If 0 the the value will be added to the message in little-endian byte order. Writes will not be reordered with regard to one another. A message written to `MSG_FIFO` one byte at a time will therefore not be affected by this setting.

### PR DESCRIPTION
**Pull request by mundaym**
The intent of this change is to make it clearer the effect that the
`kmac.CFG.msg_endianness` and `hmac.CFG.byte_swap` settings have on
values written to the relevant `MSG_FIFO`. In particular how writes
smaller than a word are affected.

This also modifies the wording for the equivalent settings for the
output registers. In this case just to clarify that each individual
word in the output has its bytes reversed, as opposed to the entire
multi-word value being reversed.

Fixes #6251.

### CODE REVIEW

File: hw/ip/hmac/data/hmac.hjson
Comment (rswarbrick) [inline hw/ip/hmac/data/hmac.hjson:72]: I wonder whether this should be "If 0 then...". Otherwise, it might be a bit difficult to break up the sentence (if there wasn't the subclause later, I'd suggest a comma: for the "If 1" case, either a comma or a "then" would probably be fine).

**Review by rswarbrick — APPROVED** [review]
This looks good, and it's nice to make the two consistent. One minor quibble about punctuation(!) and maybe it's worth letting Eunchan have a look before merging.

File: hw/ip/hmac/data/hmac.hjson
Comment (mundaym) [reply hw/ip/hmac/data/hmac.hjson]: Done ("If 0 then... If 1, ...").

**Comment by mundaym** [comment]
Thanks @rswarbrick. Yes, this should not be merged until @eunchan has reviewed and checked that the new wording is still correct.

File: hw/ip/kmac/data/kmac.hjson
Comment (udinator) [inline hw/ip/kmac/data/kmac.hjson:195]: question - maybe @eunchan can confirm, but is it worth specifying that the endian-swapping is done at a TLUL-word granularity? It looks like this is more or less stated in the below section for `state_endianness`, but IMO it's good to be even more explicit for both CSR fields as this can definitely cause some confusion.

Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:195]: Yeah. I think it is clearer to mention that in this field.

**Review by eunchan — APPROVED** [review]

File: hw/ip/kmac/data/kmac.hjson
Comment (mundaym) [reply hw/ip/kmac/data/kmac.hjson:195]: How about a sentence at the end just saying that explicitly:

`From a hardware perspective byte swaps are performed on a TL-UL word granularity.`

Does that make sense? I don't think we should assume that software developers know what TL-UL is so this sentence would be more for hardware developers.

Comment (udinator) [reply hw/ip/kmac/data/kmac.hjson:195]: I think that would be fine, thanks!

**Review by udinator — APPROVED** [review]

File: hw/ip/kmac/data/kmac.hjson
Comment (mundaym) [reply hw/ip/kmac/data/kmac.hjson:195]: Thanks! I've added that sentence to the message endianness flags.

**Comment by mundaym** [comment]
Thanks for the reviews everyone! I think all the comments have now been addressed and this can be merged once the CI tests pass.

**Review by alphan — APPROVED** [review]

**Pull request merged**
